aptos-move/framework/aptos-framework/sources/dora_committee.move

```diff
@@ -0,0 +1,1048 @@
+        );
+        assert!(
+            vector::length(&ids) == vector::length(&rpc_port),
+            error::invalid_argument(INVALID_COMMITTEE_NUMBERS)
+        );
+        while (vector::length(&ids) > 0) {
+            let id = vector::pop_back(&mut ids);
+            let node_address = vector::pop_back(&mut node_addresses);
+            let ip_public_address = vector::pop_back(&mut ip_public_address);
+            let dora_public_key = vector::pop_back(&mut dora_public_key);
+            let network_public_key = vector::pop_back(&mut network_public_key);
+            let elgamal_pub_key = vector::pop_back(&mut elgamal_pub_key);
+            let network_port = vector::pop_back(&mut network_port);
+            let rpc_port = vector::pop_back(&mut rpc_port);
+            upsert_committee_member(
+                com_store_addr,
+                owner_signer,
+                id,
+                node_address,
+                ip_public_address,
+                dora_public_key,
+                network_public_key,
+                elgamal_pub_key,
+                network_port,
+                rpc_port
+            );
+        }
+    }
+
+    /// Remove the node from the committee
+    public entry fun remove_committee_member(
+        com_store_addr: address,
+        owner_signer: &signer,
+        id: u64,
+        node_address: address
+    ) acquires CommitteeInfoStore, SupraCommitteeEventHandler {
+        // Only the OwnerCap capability can access it
+        let _acquire = &capability::acquire(owner_signer, &OwnerCap {});
+
+        let committee_store = borrow_global_mut<CommitteeInfoStore>(com_store_addr);
+        let committee = simple_map::borrow_mut(&mut committee_store.committee_map, &id);
+        ensure_node_address_exist(committee, node_address);
+        let (_, node_info) = simple_map::remove(&mut committee.map, &node_address);
+        let owner_address = signer::address_of(owner_signer);
+        let event_handler = borrow_global_mut<SupraCommitteeEventHandler>(owner_address);
+        emit_event(
+            &mut event_handler.remove_committee_member,
+            RemoveCommitteeMemberEvent {
+                committee_id: id,
+                committee_member: DoraNodeInfo {
+                    ip_public_address: node_info.ip_public_address,
+                    dora_public_key: node_info.dora_public_key,
+                    network_public_key: node_info.network_public_key,
+                    elgamal_pub_key: node_info.elgamal_pub_key,
+                    network_port: node_info.network_port,
+                    rpc_port: node_info.rpc_port,
+                }
+            }
+        );
+        // Remove the node from the node_to_committee_map
+        simple_map::remove(&mut committee_store.node_to_committee_map, &node_address);
+    }
+
+    /// The node can update its information
+    /// If the node is not here, report errors
+    public entry fun update_node_info(
```

Comment:
what is the difference between this method and `upsert_committee_member`?

Reply:
I think this is a subset of `upsert_committee_member`, let's use `upsert_committee_member` for general update for committee member